subsys/net/ip/Kconfig

```diff
@@ -139,6 +139,27 @@ config NET_TCP_ACK_TIMEOUT
 	various TCP states. The value is in milliseconds. Note that
 	having a very low value here could prevent connectivity.
 
+config NET_TCP_RETRY_COUNT
+	int "Maximum number of TCP fragment retransmissions"
+	depends on NET_TCP
+	default 10
+	help
+	The following formula can be used to determine the time (in ms)
+	that a fragment will be be buffered awaiting retransmission:
+
+	       n=NET_TCP_RETRY_COUNT
+		∑((1<<n) * 200)
+	       n=0
+
+	With the default value of 10, the IP stack will try to
```

Comment:
The 3:24min timeout sounds quite long for a device that has limited resources. I think the minimum mentioned in RFC 1122 sounds slightly better.